src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -27,7 +27,7 @@ internal sealed class MsQuicConnection : QuicConnectionProvider
 
         // TODO: remove this.
         // This is only used for client-initiated connections, and isn't needed even then once Connect() has been called.
-        private readonly SafeMsQuicConfigurationHandle? _configuration;
+        private SafeMsQuicConfigurationHandle? _configuration;
```

Comment:
Given the comment here, I wonder why we don't just create (and dispose) this in ConnectAsync so we don't have to store the value at all.

Reply:
I think we may. That would possibly change where exception can be thrown e.g. in constructor were the options are passed in or  in Connect. We would need to store all the options too, right?

Reply:
Yeah, I suppose we need to store the options as well.

Though alternatively, we could pass options to ConnectAsync and never have to store them at all. I don't think we should be making that kind of API change right now, but we can certainly consider it for 7.0.

Reply:
@wfurt I filed an issue for 7.0 to consider changing the API here to make things simpler, here: https://github.com/dotnet/runtime/issues/56009

Please take a look at the issue and make sure it looks good to you.